code/game/objects/structures/fireaxe.dm

```diff
@@ -10,62 +10,69 @@
 	integrity_failure = 0.33
 	var/locked = TRUE
 	var/open = FALSE
-	var/obj/item/fireaxe/fireaxe
+	var/obj/item/held_item
+	var/item_path = /obj/item/fireaxe
+	var/item_overlay = "axe"
 
 MAPPING_DIRECTIONAL_HELPERS(/obj/structure/fireaxecabinet, 32)
 
 /obj/structure/fireaxecabinet/Initialize(mapload)
 	. = ..()
-	fireaxe = new(src)
+	held_item = new item_path(src)
 	update_appearance()
 
 /obj/structure/fireaxecabinet/Destroy()
-	if(fireaxe)
-		QDEL_NULL(fireaxe)
+	if(held_item)
+		QDEL_NULL(held_item)
 	return ..()
 
-/obj/structure/fireaxecabinet/attackby(obj/item/I, mob/living/user, params)
-	if(iscyborg(user) || I.tool_behaviour == TOOL_MULTITOOL)
+/obj/structure/fireaxecabinet/attackby(obj/item/attacking_item, mob/living/user, params)
+	if(iscyborg(user) || attacking_item.tool_behaviour == TOOL_MULTITOOL)
 		toggle_lock(user)
-	else if(I.tool_behaviour == TOOL_WELDER && !user.combat_mode && !broken)
+	else if(attacking_item.tool_behaviour == TOOL_WELDER && !user.combat_mode && !broken)
 		if(atom_integrity < max_integrity)
-			if(!I.tool_start_check(user, amount=2))
+			if(!attacking_item.tool_start_check(user, amount = 2))
 				return
-
 			to_chat(user, span_notice("You begin repairing [src]."))
-			if(I.use_tool(src, user, 40, volume=50, amount=2))
-				atom_integrity = max_integrity
+			if(attacking_item.use_tool(src, user, 4 SECONDS, volume= 50, amount = 2))
+				repair_damage(max_integrity - get_integrity())
 				update_appearance()
 				to_chat(user, span_notice("You repair [src]."))
 		else
 			to_chat(user, span_warning("[src] is already in good condition!"))
 		return
-	else if(istype(I, /obj/item/stack/sheet/glass) && broken)
-		var/obj/item/stack/sheet/glass/G = I
-		if(G.get_amount() < 2)
+	else if(istype(attacking_item, /obj/item/stack/sheet/glass) && broken)
+		var/obj/item/stack/sheet/glass/glass_stack = attacking_item
+		if(glass_stack.get_amount() < 2)
 			to_chat(user, span_warning("You need two glass sheets to fix [src]!"))
 			return
 		to_chat(user, span_notice("You start fixing [src]..."))
-		if(do_after(user, 20, target = src) && G.use(2))
+		if(do_after(user, 2 SECONDS, target = src) && glass_stack.use(2))
 			broken = FALSE
 			atom_integrity = max_integrity
 			update_appearance()
 	else if(open || broken)
-		if(istype(I, /obj/item/fireaxe) && !fireaxe)
-			if(HAS_TRAIT(I, TRAIT_WIELDED))
-				to_chat(user, span_warning("Unwield [I] first."))
+		if(istype(attacking_item, item_path) && !held_item)
+			if(HAS_TRAIT(attacking_item, TRAIT_WIELDED))
+				to_chat(user, span_warning("Unwield [attacking_item] first."))
```

Comment:
Since this is a failure state, should it end with `!`?

Reply:
sure